Adds documentation for W&B Skills (#2258)
wandb/docs: platform/wb-skills.mdx

---
title: Use W&B Skills
description: Install W&B Skills to teach your coding agent how to train models, build agents, and analyze experiments using W&B's AI development platform.
---

W&B Skills are reusable instruction sets that teach coding agents how to use W&B effectively. Instead of manually guiding your agent through W&B APIs and best practices, install Skills so that it can work with experiment tracking, tracing, evaluations, and monitoring on its own.

> Skills work with most major coding agents, including:

Reviewer: ```suggestion
Skills work with several major coding agents, including:
```

* Claude Code
* Codex
* Cursor
* GitHub Copilot
* Gemini CLI

> For a full list of supported agents, see the [skills CLI documentation](https://github.com/vercel-labs/skills#supported-agents).

Reviewer: ```suggestion
For a full list of supported agents, see the [W&B Skills CLI documentation](https://github.com/vercel-labs/skills#supported-agents).
```
Or `skills` CLI if that's what you were going for, maybe?

## W&B Skills capabilities

Skills covers both the [W&B Models SDK](/models/ref) (training runs, metrics, artifacts, sweeps) and the [Weave SDK](/weave/reference/python-sdk) (traces, evaluations, scorers). Includes helper libraries, reference docs, and data analysis patterns.

| Workflow | Capabilities |
|----------|-------------|

> | **Model training** | <ul><li>Log metrics and rich media during training and fine-tuning.</li><li>Track and compare experiments.</li><li>Analyze runs and results, such as loss curves and accuracy scores.</li><li>Tune hyperparameters.</li></ul> |

Reviewer: Sorry about the HTML hassle but it looks great!

| **Agent building** | <ul><li>Trace agentic AI applications.</li><li>Analyze traces and classify failure modes.</li><li>Evaluate models and agents with labeled datasets.</li><li>Run online evaluations for production monitoring.</li></ul> |

## Prerequisites

Skills requires the following:

* [Node.js](https://nodejs.org/) (for the `npx` command).
* A W&B API key. Create one at [wandb.ai/authorize](https://wandb.ai/authorize) and then set it as an environment variable:
    ```shell
    export WANDB_API_KEY=<your-api-key>
    ```
* (Optional) Set your W&B project name as a `WANDB_PROJECT` environment variable. This allows your agent to target the correct W&B project without you specifying it each time.

## Install W&B Skills

To install W&B Skills globally, run the following command with the `--global` flag:

```bash

> npx skills add wandb/skills --skill '*' --yes --global

Reviewer: Should there be an example of installing specific skills? They all have `'*'`.

Author: There is only one skill right now :/

putting something in the text below about seeing the CLI reference for available skills, so that we'll have *some* guidance there.

---
title: Use W&B Skills
description: Install W&B Skills to teach your coding agent how to train models, build agents, and analyze experiments using W&B's AI development platform.
---

> W&B Skills are reusable instruction sets that teach coding agents how to use W&B effectively. Instead of manually guiding your agent through W&B APIs and best practices, install W&B Skills so that it can work with experiment tracking, tracing, evaluations, and monitoring on its own.

Reviewer: Nit: We don't tend to keep reiterating the "W&B" after the first mention. But maybe "skills" is too vague for following mentions.

Author: yeah, it felt weird to just call it Skills, but I'm gonna roll with this and see how it goes.

Skills work with most major coding agents, including:
* Claude Code
* Codex
* Cursor
* GitHub Copilot
* Gemini CLI

For a full list of supported agents, see the [skills CLI documentation](https://github.com/vercel-labs/skills#supported-agents).

## W&B Skills capabilities

W&B Skills covers both the [W&B Models SDK](/models/ref) (training runs, metrics, artifacts, sweeps) and the [Weave SDK](/weave/reference/python-sdk) (traces, evaluations, scorers). Includes helper libraries, reference docs, and data analysis patterns.

> **For model training:**

Reviewer: These might look nice as a simple two-column table.

* Log metrics and rich media during model training and fine-tuning.
* Track and compare experiments.
* Analyze runs and results, such as loss curves and accuracy scores.
* Tune hyperparameters.

**For agent building:**
* Trace agentic AI applications.
* Analyze traces and classify failure modes.
* Evaluate models and agents with labeled datasets.
* Run online evaluations for production monitoring.

## Prerequisites

W&B Skills requires the following:

* [Node.js](https://nodejs.org/) (for the `npx` command).
* A W&B API key. Create one at [wandb.ai/authorize](https://wandb.ai/authorize) and then set it as an environment variable:

>     ```

Reviewer: ```suggestion
    ```shell
```

    export WANDB_API_KEY=<your-api-key>
    ```

## Install W&B Skills

To install W&B Skills globally, run the following command with the `--global` flag:

```bash
npx skills add wandb/skills --skill '*' --yes --global
```

To install W&B Skills for a specific project, run the following command from your project directory:

```bash
npx skills add wandb/skills --skill '*' --yes
```

You can also install W&B Skills for specific agents using the `--agent` flag:

```bash
npx skills add wandb/skills --agent claude-code --skill '*' --yes --global
```

For list of `--agent` options, see the [skills CLI documentation](https://github.com/vercel-labs/skills#supported-agents).

## Use W&B Skills

> Once installed, you can ask the agent to perform W&B-related tasks for your project. The following prompts demonstrate tasks your agent can perform the skill installed:

Reviewer: ```suggestion
Once installed, you can ask the agent to perform W&B-related tasks for your project. The following example prompts demonstrate some of the tasks your agent can do with W&B Skills:
```

> * "Log training metrics for my PyTorch model to W&B."

Reviewer: Curious, is an agreed-upon way to format prompts in docs emerging yet?

Author: I haven't really seen anything, but Google is kinda just doing what we're doing: https://cloud.google.com/discover/what-is-prompt-engineering

* "Analyze the loss curves for my last 10 runs and identify the best performing configuration."
* "Trace my LangChain agent and log the results to Weave."
* "Run an evaluation on my agent using the test dataset and summarize the results."
* "Find the failure modes in my last evaluation and classify them."
* "Compare the configs of run A and run B and show me the differences."

## Usage tips

> - **Set your W&B project as a `WANDB_PROJECT` environment variable**: This allows your agent to target the correct W&B project without you specifying it each time.

Reviewer: Nit: I'd tighten this up a bit and align with other mentions of setting this variable.

> - **Ask specific questions**: Instead of "how is my model doing?", ask "what is the final validation loss for my last 5 runs?"

Reviewer: This might be good as a table of too-vague prompts and suggested more specific ones.

description: Install W&B Skills to teach your coding agent how to train models, build agents, and analyze experiments using W&B's AI development platform.
---

> W&B Skills are reusable instruction sets that teach coding agents how to use W&B effectively. Instead of manually guiding your agent through W&B APIs and best practices, install Skills so that it can work with experiment tracking, tracing, evaluations, and monitoring on its own.

Reviewer: Maybe "so that the agent can work"
